service/awsconfig/v10/resource/endpoints/desired.go

```diff
@@ -12,17 +13,28 @@ import (
 
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
 	customObject, err := key.ToCustomObject(obj)
-	if IsNotFound(err) {
-		// Fall through.
-		return nil, nil
-	}
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	instanceName := key.MasterInstanceName(customObject)
 	masterInstance, err := r.findMasterInstance(instanceName)
-	if err != nil {
+	if IsNotFound(err) {
+		// During updates the master instance is shut down and thus cannot be found.
+		// In such cases we cancel the reconciliation for the endpoint resource.
+		// This should be ok since all endpoints should be created and up to date
+		// already. In case we miss an update it will be done on the next resync
+		// period once the master instance is up again.
+		//
+		// TODO we might want to alert at some point when the master instance was
+		// not seen for too long. Like we should be able to find it again after
+		// three resync periods max or something.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "master instance not found")
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource reconciliation for custom object")
+
+		return nil, nil
+	} else if err != nil {
```

Comment:
Figured this has to be more graceful due to repeated delete events caused by finalizers and the updates themselves.